src/rules/no-reducer-in-key-names.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils'
+
+import { actionReducerMap, docsUrl, storeActionReducerMap } from './utils'
+
+export const ruleName = 'no-reducer-in-key-names'
+
+export const messageId = 'noReducerInKeyNames'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: ruleName,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Possible Errors',
+      description: 'Avoid the word "reducer" in the key names',
+      recommended: 'error',
+    },
+    schema: [],
+    messages: {
+      [messageId]:
+        'Avoid the word "reducer" in the key names to better represent the state.',
+    },
+  },
+  defaultOptions: [],
+  create: context => {
+    const excludedKeyword = 'reducer'
+
+    function hasKeyword(name: string): boolean {
+      return name.toLowerCase().includes(excludedKeyword)
+    }
+
+    return {
+      [`${storeActionReducerMap}, ${actionReducerMap}`](
+        node: TSESTree.Property,
+      ) {
+        const key = node.key
+        if (
+          (key.type === AST_NODE_TYPES.Literal &&
```

Comment:
I would prefer this check implemented as a guard (similar to https://github.com/timdeschryver/eslint-plugin-ngrx/blob/v1.0.0/src/rules/utils/helper-functions/index.ts#L3)
It will have as result that the node will have the correct type, thus `key.value as string` would not be needed normally.

Thoughts?

Reply:
I implemented the guard because it's more readable. However key.value on a literal can be many things so `key.value as string` is still necessary
![image](https://user-images.githubusercontent.com/49377434/83352432-97a0e480-a34b-11ea-820a-de1481b0cd0a.png)

Reply:
It seems I can use raw property which is already a string. So that's what I've done

Reply:
👍